Bowling_Calculator/BowlingCalculator.cs

```diff
@@ -0,0 +1,283 @@
+                UpdateTotalScoreOfFrame(frame_id_counter - 1);
+            }
+        }
+
+        public List<int> Get_frame_result(int frame_id)
+        {
+            return results[frame_id - 1].Get_resultThrows();
+        }
+
+        public int Get_frame_score(int frame_id)
+        {
+            return results[frame_id - 1].Get_frameScore();
+        }
+
+        public int Get_total(int frame_id)
+        {
+            return results[frame_id - 1].Get_total();
+        }
+
+        public int Get_latest_total_score()
+        {
+            return results[results.Count - 1].Get_total();
+        }
+
+        class Frame
+        {
+            int id;
+            bool isStrike;
+            bool isSpare;
+            bool isBonusThrow;
+            bool isFrameEnded;
+            List<int> resultThrows = new List<int>();
+            List<int> addedThrows = new List<int>();
+            int frameScore;
+            int total;
+
+            public Frame(int id)
+            {
+                this.id = id;
+                this.isStrike = false;
+                this.isSpare = false;
+                this.isBonusThrow = false;
+                this.isFrameEnded = false;
+                this.frameScore = 0;
+                this.total = 0;
+            }
+
+            public void Add_FrameScore(int resultThrow)
+            {
+                this.frameScore += resultThrow;
+                this.addedThrows.Add(resultThrow);
+            }
+
+            public void Add_resultThrow(int resultThrow)
+            {
+                this.resultThrows.Add(resultThrow);
+                this.Add_FrameScore(resultThrow);
+                this.total += resultThrow;
+                if (this.isFrameEnded == false)
+                {
+                    if ((resultThrow == 10))
+                    {
+                        this.isStrike = true;
+                        if (!this.isBonusThrow) { }
+                        this.isFrameEnded = true;
+                    }
+                    else if (addedThrows.Count == 2)
+                    {
+                        if ((addedThrows[0] + addedThrows[1]) == 10)
+                        {
+                            this.isSpare = true;
+                        }
+                        this.isFrameEnded = true;
+                    }
+                    // spare bonus end
+                    if ((this.isBonusThrow) && (this.isSpare) && (this.addedThrows.Count == 3))
+                    {
+                        this.isFrameEnded = true;
+                    }
+                    // strike bonus end
+                    else if ((this.isBonusThrow) && (this.isStrike) && (this.addedThrows.Count == 3))
+                    {
+                        this.isFrameEnded = true;
+                    }
+                    else if (this.isBonusThrow)
+                    {
+                        this.isFrameEnded = false;
+                    }
```

Comment:
This could be converted to switch-case statement (yes it isn't python, we have switch-case :-) here) 
But it still strange, may we could reduce this chain?

Reply:
I don't think switch-case will be usefull here as all three conditions do not follow the same pattern. Though the chain can be reduced by combining first two expressions, thank you.